types/underscore/index.d.ts

```diff
@@ -3597,6 +3597,22 @@ declare module _ {
             context?: any
         ): Extract<keyof V, string> | undefined;
 
+        /**
+         * Returns the specified property of `object`. `path` may be specified
+         * as a simple key, or as an array of object keys or array indexes,
+         * for deep property fetching. If the property does not exist or is `undefined`,
+         * the optional default is returned.
+         * @param object The object that maybe contains the property.
+         * @param path The path to the property on `object`.
+         * @param defaultValue Default if not found.
+         * @returns The item on the `object` or the `defaultValue`
+         **/
+        get(
+            object: object,
+            path: string[],
+            defaultValue: any
+        ): any;
```

Comment:
You can make the return type much more sensitive to situations where TS is able to infer it. This requires some juggling with helper types, generics and overloads. You can take example from the existing typings for `property` and `propertyOf`.